### PR TITLE
test: create E2E test for Next.js and CSS extraction

### PR DESCRIPTION
**Pull request by layershifter**
This PR adds E2E test for Next.js plugin to ensure that CSS extraction works as expected.

- Some utils were moved out from TS into `@griffel/e2e-utils` to share them between tests
- Tests and utils were rewritten to TS for better expirence
- `swc-node` was added to run TS code
- `log-symbols` was downgraded as it requires ESM, but `.mts` is not supported by SWC https://github.com/swc-project/swc/issues/4679

### CODE REVIEW

File: e2e/typescript/tsconfig.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
     "allowJs": true,
     "checkJs": true,
     "module": "CommonJS",
+    "lib": ["ES2019", "DOM", "DOM.Iterable"],
```
Comment (layershifter) [inline e2e/typescript/tsconfig.json:7]: To use `Object.fromEntries`. It seems that we just need to enable `ES2019` everywhere. I will do it in a separate PR.

File: tsconfig.base.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
       "@griffel/babel-preset": ["packages/babel-preset/src/index.ts"],
       "@griffel/core": ["packages/core/src/index.ts"],
       "@griffel/devtools": ["packages/devtools/src/index.ts"],
-      "@griffel/e2e-typescript": ["e2e/typescript/src/index.ts"],
```
Comment (layershifter) [inline tsconfig.base.json:21]: These is no sense to have `@griffel/e2e-typescript` there as it exports nothing.

File: e2e/nextjs/src/assets/next.config.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+const { withGriffelCSSExtraction } = require('@griffel/next-extraction-plugin');
```
[the remark this answers is not in the document]
Comment (ling1726) [reply e2e/nextjs/src/assets/next.config.js:1]: you could just add that as a comment to the file?

[the remark this answers is not in the document]
Comment (ling1726) [reply e2e/nextjs/src/assets/next.config.js:1]: or README

File: e2e/utils/src/installPackages.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  if (packages.length > 0) {
+    const yarnOutput = await sh(`yarn info ${packages.join(' ')} --json`, rootDir, true);
+    const parsedYarnOutput = yarnOutput
+      .split('\n')
+      .filter(Boolean)
+      .map(line => JSON.parse(line));
+
+    packagesWithVersions = Object.fromEntries(
+      parsedYarnOutput.map(info => [info.value.split('@patch')[0].split('@npm')[0], info.children.Version]),
```
Comment (ling1726) [inline e2e/utils/src/installPackages.ts:32]: what kind of package would have `@patch`?

**Review by ling1726 — APPROVED** [review]

File: e2e/nextjs/src/assets/next.config.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (layershifter) [reply e2e/nextjs/src/assets/next.config.js:1]: Added

File: e2e/utils/src/installPackages.ts (hunk 4 above)
Comment (layershifter) [reply e2e/utils/src/installPackages.ts:32]: image

Yarn applies a patch to TS.

Comment (layershifter) [reply e2e/utils/src/installPackages.ts:32]: I will probably cover some of these utils with tests in upcoming PRs.

**Pull request merged**